src/olcs/Service.js

```diff
@@ -15,6 +44,17 @@ const Service = class {
    */
   initialize(manager) {
     this.manager_ = manager;
+
+    const unWatchFn = this.$rootScope.$watch(
+      () => this.manager_.getOl3d(),
```

Comment:
You should use the promise returned by `manager.load()`.

Reply:
If i do this, cesium will be loaded on application bootstrap, cause calling `load()` function from `intialize` will trigger cesium loading, doesn't it ?

Reply:
Sure, you need to check for the presence of the 3d parameters in the URL first.

Reply:
We can discuss it later, but the way you managed the load promise cannot work, we have to use a $watcher for now.